pkg/operation/botanist/controlplane.go

```diff
@@ -470,3 +474,80 @@ func (b *Botanist) WakeUpControllers() error {
 	}
 	return nil
 }
+
+// DeployCertBroker deploys the Cert-Broker to the Shoot namespace in the Seed.
+func (b *Botanist) DeployCertBroker() error {
+	certificateManagement, ok := b.Operation.Secrets[common.GardenRoleCertificateManagement]
+	if !ok {
+		return fmt.Errorf("certificate management is enabled but no secret with role %s could be found", common.GardenRoleCertificateManagement)
+	}
+	certificateManagementConfig, err := certmanagement.RetrieveCertificateManagementConfig(certificateManagement)
+	if err != nil {
+		return fmt.Errorf("certificate management configuration could not be created %v", err)
+	}
+
+	dns := []interface{}{}
+	for _, route53Provider := range certificateManagementConfig.Providers.Route53 {
+		createDNSProviders(&route53Provider, &dns)
+	}
+	for _, cloudDNSProvider := range certificateManagementConfig.Providers.CloudDNS {
+		createDNSProviders(&cloudDNSProvider, &dns)
+	}
+
+	certBrokerConfig := map[string]interface{}{
+		"replicas": b.Shoot.GetReplicas(1),
+		"certbroker": map[string]interface{}{
+			"namespace":           b.Shoot.SeedNamespace,
+			"targetClusterSecret": common.CertBrokerResourceName,
+		},
+		"certmanager": map[string]interface{}{
+			"clusterissuer": certificateManagementConfig.ClusterIssuerName,
+			"dns":           dns,
+		},
+		"podAnnotations": map[string]interface{}{
+			"checksum/secret-cert-broker": b.CheckSums[common.CertBrokerResourceName],
+		},
+	}
+
+	certBroker, err := b.InjectImages(certBrokerConfig, b.K8sSeedClient.Version(), b.K8sSeedClient.Version(), common.CertBrokerImageName)
+	if err != nil {
+		return nil
+	}
+
+	return b.ApplyChartSeed(filepath.Join(chartPathControlPlane, "cert-broker"), "cert-broker", b.Shoot.SeedNamespace, nil, certBroker)
+}
+
+func createDNSProviders(config certmanagement.DNSProviderConfig, existingDNSConfig *[]interface{}) {
```

Comment:
don't mutate the `existingDNSConfig` here, rather return a `[]interface` array and append at the place where you're using this function.